pages/api/users/[username]/links/[url].js

```diff
@@ -1,26 +1,41 @@
-import Link from "../../../models/Link";
-import Profile from "../../../models/Profile";
-import Stats from "../../../models/Stats";
-import connectMongo from "../../../config/mongo";
+import fs from "fs";
+import path from "path";
+
+import Link from "../../../../../models/Link";
+import Profile from "../../../../../models/Profile";
+import Stats from "../../../../../models/Stats";
+import connectMongo from "../../../../../config/mongo";
 
 export default async function handler(req, res) {
   await connectMongo();
-  const { data } = req.query;
-  const username = data[0];
-  const url = data[1];
 
-  if (req.method != "PUT" || !username || !url) {
+  const { username, url } = req.query;
+
+  if (req.method != "GET") {
     return res
       .status(400)
-      .json({ error: "Invalid request: 'username' and url 'required'" });
+      .json({ error: "Invalid request: GET request required" });
+  }
+
+  // load profile json file and check link
+  const filePath = path.join(process.cwd(), "data", `${username}.json`);
+  let data = {};
+  try {
+    data = JSON.parse(fs.readFileSync(filePath, "utf8"));
+  } catch (e) {
+    console.log(`ERROR loading username ${username}`, e);
+    return res.status(404).json({ error: `ERROR ${username} not found` });
+  }
+
+  if (!data.links.find((link) => link.url === url)) {
+    console.log(`ERROR link ${url} not found for username ${username}`);
```

Comment:
We also use this endpoint for socials 😅

Reply:
Good spot 👍 

(oops I hit resolved instead of comment, time for me to take a break)

Reply:
I fixed with a `&&` condition, please can you re-review